comp/core/telemetry/telemetry.go

```diff
@@ -0,0 +1,200 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"net/http"
+
+	promOtel "go.opentelemetry.io/otel/exporters/prometheus"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+)
+
+// TODO (components): Remove the global and move this into `newTelemetry` after all telemetry is migrated to the component
+var (
+	registry = newRegistry()
+	provider = newProvider(registry)
+)
+
+type telemetryImpl struct {
+	registry      *prometheus.Registry
+	meterProvider *sdk.MeterProvider
+}
+
+func newRegistry() *prometheus.Registry {
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	reg.MustRegister(collectors.NewGoCollector())
+	return reg
+}
+
+func newProvider(reg *prometheus.Registry) *sdk.MeterProvider {
+	exporter, err := promOtel.New(promOtel.WithRegisterer(reg))
+
+	if err != nil {
+		panic(err)
```

Comment:
If [provider](https://github.com/DataDog/datadog-agent/pull/17596/files#diff-105ee26f9d7031dfe68b182c6090d192717eb0d1637c558922053fcf79f1a10bR23) can be a non global state, then you can return an error instead.